### ISSUE TITLE
Flaky maven-checks job due to plugin not found

### ISSUE THREAD

**Issue by ebyhr** [trinodb/trino#11054]
```
2022-02-16T01:46:48.596Z	ERROR	main	io.trino.server.Server	io.trino.spi.Plugin: Provider io.trino.plugin.geospatial.GeoPlugin not found
java.util.ServiceConfigurationError: io.trino.spi.Plugin: Provider io.trino.plugin.geospatial.GeoPlugin not found
	at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:589)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass(ServiceLoader.java:1212)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1221)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1265)
	at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1300)
	at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1385)
	at com.google.common.collect.ImmutableList.copyOf(ImmutableList.java:275)
	at com.google.common.collect.ImmutableList.copyOf(ImmutableList.java:239)
	at io.trino.server.PluginManager.loadPlugin(PluginManager.java:164)
	at io.trino.server.PluginManager.loadPlugin(PluginManager.java:155)
	at io.trino.server.ServerPluginsProvider.lambda$loadPlugins$1(ServerPluginsProvider.java:58)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:31)
	at java.base/java.util.concurrent.ExecutorCompletionService.submit(ExecutorCompletionService.java:184)
	at io.trino.util.Executors.executeUntilFailure(Executors.java:41)
	at io.trino.server.ServerPluginsProvider.loadPlugins(ServerPluginsProvider.java:53)
	at io.trino.server.PluginManager.loadPlugins(PluginManager.java:135)
	at io.trino.server.Server.doStart(Server.java:126)
	at io.trino.server.Server.lambda$start$0(Server.java:80)
	at io.trino.$gen.Trino_501ee7a____20220216_014637_1.run(Unknown Source)
	at io.trino.server.Server.start(Server.java:80)
	at io.trino.server.TrinoServer.main(TrinoServer.java:38)
```

https://github.com/trinodb/trino/runs/5209697875

**Comment by ebyhr** [comment]
https://github.com/trinodb/trino/runs/5668476656

**Comment by nineinchnick** [comment]
I think we should close this. There was this error in logs:
```
Warning: ][warning][gc,alloc] pool-571-thread-2: Retried waiting for GCLocker too often allocating 625002 words
Terminating due to java.lang.OutOfMemoryError: Java heap space
```
which might have produced a corrupted artifact, and this should be addressed by #11593

**Comment by hashhar** [comment]
@nineinchnick That explains why it occurred most often with JDK 17 job (but on others as well).
